app.go

```diff
@@ -461,6 +473,20 @@ func (app *App) StopTimeout() time.Duration {
 	return app.stopTimeout
 }
 
+// DotGraph provides a graph of dependencies in the container as a string in DOT
+// syntax.
+func (app *App) DotGraph() DotGraph {
+	// Memoize the graph
+	if app.dotgraph != nil {
+		return *app.dotgraph
+	}
+	b := &bytes.Buffer{}
+	dig.Visualize(app.container, b) // TODO: handle errors gracefully
```

Comment:
RE: error handling: `fx.Provide`s can just return errors to break the container. Although in our case, we'll maybe want to provide an empty graph with an error message or something rather than fail application startup.

Reply:
I like the empty graph suggestion.  I think I will go with that.